src/Strategy/Swift/PackageSwift.hs

```diff
@@ -0,0 +1,248 @@
+      try (symbol ("." <> t) *> betweenBrackets parseQuotedText)
+        <|> parseKeyValue t parseQuotedText
+
+    parseUpToOperator :: Text -> Parser Text
+    parseUpToOperator t = symbol ("." <> t) *> betweenBrackets (parseRequirement "from")
+
+    parseRange :: Text -> Parser (Text, Text)
+    parseRange rangeOperator = do
+      lhs <- parseQuotedText
+      _ <- symbol rangeOperator
+      rhs <- parseQuotedText
+      pure (lhs, rhs)
+
+    optionallyTry :: ParsecT Void Text Identity a -> ParsecT Void Text Identity (Maybe a)
+    optionallyTry p = optional . try $ p <* maybeComma
+
+    parseGitDep :: Parser SwiftPackageDep
+    parseGitDep = do
+      _ <- symbol ".package" <* symbol "("
+      _ <- optionallyTry (parseKeyValue "name" parseQuotedText)
+
+      -- Url (Required Field)
+      url <- parseKeyValue "url" $ parseQuotedText <* maybeComma
+
+      -- Version Constraint (Optional Fields)
+      revision <- optionallyTry $ parseRequirement "revision"
+      branch <- optionallyTry $ parseRequirement "branch"
+      from <- optionallyTry $ parseRequirement "from"
+      exact <- optionallyTry $ parseRequirement "exact"
+      upToMajor <- optionallyTry $ parseUpToOperator "upToNextMajor"
+      upToMinor <- optionallyTry $ parseUpToOperator "upToNextMinor"
+      closedInterval <- optionallyTry $ parseRange "..."
+      rhsHalfOpenRange <- optionallyTry $ parseRange "..<"
+
+      _ <- symbol ")"
+      pure $
+        GitSource $
+          SwiftPackageGitDep
+            url
+            branch
+            revision
+            from
+            exact
+            upToMajor
+            upToMinor
+            closedInterval
+            rhsHalfOpenRange
+
+parsePackageDependencies :: Parser [SwiftPackageDep]
+parsePackageDependencies = do
+  _ <- lexeme $ skipManyTill anySingle $ symbol "let package = Package("
+  skipManyTill anySingle (symbol "dependencies:") *> betweenSquareBrackets (sepEndBy (lexeme parsePackageDep) $ symbol ",")
+
+parseSwiftToolVersion :: Parser Text
+parseSwiftToolVersion =
+  symbolWOComment "//"
+    *> parseKeyValue
+      "swift-tools-version"
+      (takeWhile1P (Just "swift-tools-version") $ not . isEndLine)
+
+parsePackageSwiftFile :: Parser SwiftPackage
+parsePackageSwiftFile = do
+  -- Package.swift must specify version for swift tools
+  -- https://github.com/apple/swift-package-manager/blob/main/Documentation/PackageDescription.md#about-the-swift-tools-version
+  swiftToolVersion <- parseSwiftToolVersion
+  SwiftPackage swiftToolVersion <$> parsePackageDependencies
+
+-- | Analysis
+-- *
+analyzePackageSwift :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m (Graphing.Graphing Dependency, GraphBreadth)
+analyzePackageSwift manifestFile = do
+  manifestContent <- context "Identifying dependencies in Package.swift" $ readContentsParser parsePackageSwiftFile manifestFile
+  graph <- context "Building dependency graph" $ pure $ buildGraph manifestContent
+  pure (graph, Partial)
```

Comment:
If this is always partial, then you should remove it from this function and just use it higher up.